client/src/composables/toolTrainingMaterial.ts

```diff
@@ -0,0 +1,168 @@
+import { useConfig } from "./config";
+import { computed, ref, watch, type Ref } from "vue";
+
+type Name = string;
+type Repo = string;
+type ID = string;
+
+type TrainingId = `${Repo}/${Name}` | ID;
+
+type TrainingDetails = {
+    tool_id: Array<
+        [
+            string, // ID (unused)
+            string // Version
+        ]
+    >;
+    tutorials: Array<
+        [
+            string, // tutorial ID (unused)
+            string, // Title
+            string, // Category
+            string // URL
+        ]
+    >;
+};
+
+type TrainingMaterialResponse = {
+    [id: TrainingId]: TrainingDetails;
+};
+
+type Config = {
+    tool_training_recommendations: boolean;
+    tool_training_recommendations_api_url: string;
+    tool_training_recommendations_link: string;
+};
+
+export type TutorialDetails = {
+    category: string;
+    title: string;
+    url: URL;
+};
+
+const cachedResponse: Ref<TrainingMaterialResponse | null> = ref(null);
+
+export function useToolTrainingMaterial(id: ID, name: Name, version: string, repo?: Repo) {
```

Comment:
name is the tool name I think, which doesn't have to appear in the tool id, I think that's why bwa-mem doesn't show any tutorials